ProgressBar with StyleBoxFlat styles has larger minimum size than in v3.0.6

<!-- Please search existing issues for potential duplicates before filing yours:
https://github.com/godotengine/godot/issues?q=is%3Aissue
-->

**Godot version:**
3.1 alpha 1


**OS/device including version:**
Ubuntu 18.04.1


**Issue description:**
Can't set `ProgressBar` `Rect Size` in the property inspector to small values. In 3.0 I could set the x and y to values as small as 0 pixels. The minimum x,y now seem to be 4 and 14. I'm using this for enemy health bar indicators. 


**Steps to reproduce:**
- Create a `ProgressBar`
- Change Fg and Bg Custom Styles to `StyleBoxFlat`
- Use the property inspector to set the `Rect Size y` property to 5 pixels. Or just try to resize the `ProgressBar` height using its handles.

It is still reproducible in beta 2. I can't build master 8ff00ca... I get this error:

```
drivers/unix/os_unix_global_settings_path.gen.cpp:3:44: error: no 'String OS_Unix::get_global_settings_path() const' member function declared in class 'OS_Unix'
 String OS_Unix::get_global_settings_path() const {
                                            ^~~~~
scons: *** [drivers/unix/os_unix_global_settings_path.gen.x11.tools.64.o] Error 1
scons: building terminated because of errors.
```

> `drivers/unix/os_unix_global_settings_path.gen.cpp:3:44: error: no 'String OS_Unix::get_global_settings_path() const' member function declared in class 'OS_Unix'`

Delete the `.gen.cpp` file.

--- a/editor/plugins/animation_blend_tree_editor_plugin.cpp
+++ b/editor/plugins/animation_blend_tree_editor_plugin.cpp
@@ -231,6 +231,7 @@ void AnimationNodeBlendTreeEditor::_update_graph() {
 			}
 
 			pb->set_percent_visible(false);
+			pb->set_custom_minimum_size(Vector2(0, 14) * EDSCALE);
 			animations[E->get()] = pb;
 			node->add_child(pb);
 

--- a/scene/gui/progress_bar.cpp
+++ b/scene/gui/progress_bar.cpp
@@ -39,9 +39,12 @@ Size2 ProgressBar::get_minimum_size() const {
 	Size2 minimum_size = bg->get_minimum_size();
 	minimum_size.height = MAX(minimum_size.height, fg->get_minimum_size().height);
 	minimum_size.width = MAX(minimum_size.width, fg->get_minimum_size().width);
-	//if (percent_visible) { this is needed, else the progressbar will collapse
-	minimum_size.height = MAX(minimum_size.height, bg->get_minimum_size().height + font->get_height());
-	//}
+	if (percent_visible) {
+		minimum_size.height = MAX(minimum_size.height, bg->get_minimum_size().height + font->get_height());
+	} else { // this is needed, else the progressbar will collapse
+		minimum_size.width = MAX(minimum_size.width, 1);
+		minimum_size.height = MAX(minimum_size.height, 1);
+	}
 	return minimum_size;
 }
 

